### ISSUE TITLE
[Feature Request] Native Request Batching to Prevent "received message larger than max" Errors

### ISSUE THREAD

**Issue by dp-dandy** [temporalio/sdk-typescript#1499]
### Is your feature request related to a problem? Please describe.
When kicking off a large amount of activities or a small amount of activities with very rich parameters. A "silent" (in the context of the node.js runtime) error occurs, where the underlying grpc request made to temporal cloud fails. This failure is currently only logged and not raised in the context of a workflow trying to kick off the associated activities.

```
[2m2024-08-15T04:31:54.812550Z[0m [33m WARN[0m [2mtemporal_sdk_core::worker::workflow[0m[2m:[0m Error while completing workflow activation [3merror[0m[2m=[0mstatus: ResourceExhausted, message: "grpc: received message larger than max (15224744 vs. 4194304)", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc", "server": "temporal", "date": "Thu, 15 Aug 2024 04:31:54 GMT"} } 
```

### Describe the solution you'd like

1. This error should be raised and more apparent to users of the sdk
    - This is a runtime error, and its an error that devs should be able to catch and handle automatically
2. The sdk should be capable of automatically grouping activity requests such that the grpc calls are not (if possible) over the maximum allowed size

A rough incomplete solution that we are piloting is something like:
```typescript
import type { ActivityFunction } from '@temporalio/common';
import { DefaultPayloadConverter } from '@temporalio/common';

const MAX_BATCH_PAYLOAD_SIZE = 3 * 1024 * 1024; // 3MB, leaving a buffer for any unknown overhead associated with the activities

const defaultPayloadConverter: DefaultPayloadConverter = new DefaultPayloadConverter();

function getPayloadSize<PayloadType extends Parameters<ActivityFunction>>(args: PayloadType): number {
    const payload = defaultPayloadConverter.toPayload(args);
    if (payload.data instanceof Uint8Array) {
        return payload.data.length;
    }
    return 0;
}

function groupIntoMaxSizePayloads<PayloadType extends Parameters<ActivityFunction>>(
    activityArgs: PayloadType[],
    maxSize: number = MAX_BATCH_PAYLOAD_SIZE,
): PayloadType[][] {
    const payloadGroups: PayloadType[][] = [];

    let currentGroup: PayloadType[] = [];
    let currentGroupSize = 0;
    for (const args of activityArgs) {
        const argsSize = getPayloadSize(args);
        if (argsSize > maxSize) {
            // throw some error about the request size being too big
        }
        // If the current group size plus the size of the next args is greater than the max size,
        // push the current group and start a new one
        if (currentGroupSize + argsSize > maxSize) {
            payloadGroups.push(currentGroup);
            currentGroup = [];
            currentGroupSize = 0;
        }
        currentGroup.push(args);
        currentGroupSize += argsSize;
    }
    // Push the last group if it's not empty
    if (currentGroup.length > 0) {
        payloadGroups.push(currentGroup);
    }
    return payloadGroups;
}

export async function safeStartActivities<Activity extends ActivityFunction, Arguments>(
    activity: Activity,
    args: Arguments[],
): Promise<PromiseSettledResult<Awaited<ReturnType<Activity>>>[]> {
    const maxRequestSizeBatches: Parameters<Activity>[][] = groupIntoMaxSizePayloads(
        args.map((arg: Arguments) => arg as Parameters<Activity>),
    );
    const results: PromiseSettledResult<Awaited<ReturnType<Activity>>>[] = [];
    for (const maxRequestSizeBatch of maxRequestSizeBatches) {
        const promises: Promise<ReturnType<Activity>>[] = maxRequestSizeBatch.map((args: Parameters<Activity>) =>
            activity(args),
        );
        const awaited = await Promise.allSettled(promises);
        results.push(...awaited);
    }
    return results;
}

```

### Additional context
Cloud Support Thread: https://temporalio.slack.com/archives/C046BRWDV2R/p1723743294696369
